test/test_jit.py

```diff
@@ -9203,6 +9203,47 @@ def nest_for_ret(x):
                         return 4
                 return 5
 
+    def test_overloading(self):
+        @torch._jit_internal.weak_module
+        class W(torch.nn.Module):
+            def __init__(self):
+                super(W, self).__init__()
+                torch._jit_internal.overload(type(self), 'forward', (self.forward_tensor, self.forward_tuple))
```

Comment:
`self.forward_tensor` and `self.forward_tuple` are not the same functions across different instances of class W. There may be different assignments to` __constant__` attributes and submodules. Caching them on the type is a bug.